src/routes/dashboard/dashboard.jsx

```diff
@@ -50,17 +54,32 @@ const Dashboard = () => {
         });
       }
     );
+    setWatchListSymbolStreamRef([
+      ...watchlist_symbol_stream_ref,
+      unsubscribeRef,
+    ]);
   };
 
   onMount(() => {
     if (!network_status.is_disconnected) {
-      const getFavs = JSON.parse(localStorage.getItem("favourites"));
+      const active_user = localStorage.getItem("userId") ?? "guest";
+      const getFavs = JSON.parse(
+        localStorage.getItem(`${active_user}-favourites`)
+      );
```

Comment:
This code is duplicated in multiple places . can we have a common function for this? will it work